### DISCUSSION TITLE
convert editor source code from CPP to GDScript

### DISCUSSION THREAD

**Discussion by shutaozhenzhen** [godotengine/godot-proposals#9692] [Editor]
Look at the main repo, I find godot editor seems building by C++. Why not using GDScript to rewrite it.

**Comment by AThousandShips** [comment]
GDScript is not a general purpose programming language, and this would be far far slower than using c++ for a bunch of reasons. There's no need or reason to write the editor in GDScript, you don't need to have GDScript activated to run the editor, or projects, you can just use C# or any other scripting language

The question is:
Why would you write the editor in GDScript, what benefits would it add, and how would they be worth the editor being slower, and harder to maintain

**Comment by shutaozhenzhen** [comment]
> What user interface toolkit does Godot use?
> 
> Godot does not use a standard GUI toolkit like GTK, Qt or wxWidgets. Instead, Godot uses its own user interface toolkit, rendered using OpenGL ES or Vulkan. This toolkit is exposed in the form of Control nodes, which are used to render the editor (which is written in C++). These Control nodes can also be used in projects from any scripting language supported by Godot.
> 
> This custom toolkit makes it possible to benefit from hardware acceleration and have a consistent appearance across all platforms. On top of that, it doesn't have to deal with the LGPL licensing caveats that come with GTK or Qt. Lastly, this means Godot is "eating its own dog food" since the editor itself is one of the most complex users of Godot's UI system.
> 
> This custom UI toolkit can't be used as a library, but you can still use Godot to create non-game applications by using the editor.

I think using GDScript to implement the Godot Editor can provide an example of what GDScript can do with Godot, and in the process of implementation can see how GDScript actually needs to be adjusted.

In short, I want the Godot Editor to be closer to Godot's current development ecosystem.

**Reply by AThousandShips** [reply]
The editor is far too complex to support that, it needs to handle low level features that aren't available to GDScript, not to mention the performance would be far too low

There has been discussions of designing parts of the UI elements in GDScript but that's very different, and even that has been rejected at least partially because of the fact that the people who work on the editor know c++, they don't necessarily know GDScript well enough to design things even when they are possible to design in GDScript

So this would only really serve the purpose of showcasing what GDScript *might* be able to do, and poorly, and would make developing and maintaining the engine much harder
